backends/s3/credentials.go

```diff
@@ -0,0 +1,48 @@
+package s3
+
+import (
+	"os"
+	"time"
+
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// FileSecretsCredentials is an implementation of Minio's credentials.Provider,
+// allowing to read credentials from Kubernetes or Docker secrets, as described in
+// https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure
+// and https://docs.docker.com/engine/swarm/secrets.
+type FileSecretsCredentials struct {
+	credentials.Expiry
+
+	// Path to the field containing the access key,
+	// e.g. /etc/s3-secrets/access-key.
+	AccessKeyFile string
+
+	// Path to the field containing the secret key,
+	// e.g. /etc/s3-secrets/secret-key.
+	SecretKeyFile string
+}
+
+// Retrieve implements credentials.Provider.
+// It reads files pointed to by p.AccessKeyFilename and p.SecretKeyFilename.
+func (c *FileSecretsCredentials) Retrieve() (credentials.Value, error) {
+	keyId, err := os.ReadFile(c.AccessKeyFile)
+	if err != nil {
+		return credentials.Value{}, err
+	}
+	secretKey, err := os.ReadFile(c.SecretKeyFile)
+	if err != nil {
+		return credentials.Value{}, err
+	}
+
+	creds := credentials.Value{
+		AccessKeyID:     string(keyId),
+		SecretAccessKey: string(secretKey),
+	}
+
+	c.SetExpiration(time.Now().Add(time.Second), -1)
```

Comment:
Let's make this configurable with a minimum of 1s, and perhaps a longer default like 15s, since Kubernetes secrets are only refreshed every minute if I understood correctly.

A slightly higher default makes it easier to detect overly strict rollovers (not keeping the previously key around), which would also be problematic with 1s, but harder to detect. This will also keep battery usage lower when used on mobile devices.